exchanges/gateio/gateio_websocket_test.go

```diff
@@ -0,0 +1,176 @@
+			continue
+		}
+		require.NoError(t, err, "processSpotBalances must not error")
+		require.Len(t, e.Websocket.DataHandler, 1)
+		payload := <-e.Websocket.DataHandler
+		got, ok := payload.([]account.Change)
+		require.Truef(t, ok, "Expected account changes, got %T", payload)
+		checkAccountChange(t, tc.expected, got, e, tc.ctx)
+	}
+}
+
+func TestProcessBalancePushData(t *testing.T) {
+	t.Parallel()
+	e := new(Exchange) //nolint:govet // Intentional shadow
+	e.SetDefaults()
+	e.Name = "ProcessFuturesBalancesTest"
+
+	ctx := account.DeployCredentialsToContext(t.Context(), &account.Credentials{Key: "test", Secret: "test"})
+
+	for _, tc := range []struct {
+		input    []byte
+		err      error
+		ctx      context.Context
+		expected []account.Change
+	}{
+		{
+			ctx:   t.Context(),
+			input: []byte(`[{"timestamp":"1755718222"}]`),
+			err:   exchange.ErrCredentialsAreEmpty,
+		},
+		{
+			ctx:   ctx,
+			input: []byte(`[{"balance":2214.191673190433,"change":-0.0025776,"currency":"usdt","text":"TCOM_USDT:263179103241933596","time":1755738515,"time_ms":1755738515671,"type":"fee","user":"12870774"}]`),
+			expected: []account.Change{
+				{
+					Account:   "12870774",
+					AssetType: asset.USDTMarginedFutures,
+					Balance: &account.Balance{
+						Currency:               currency.USDT,
+						Total:                  2214.191673190433,
+						Free:                   2214.191673190433,
+						AvailableWithoutBorrow: 2214.191673190433,
+						UpdatedAt:              time.UnixMilli(1755738515671),
+					},
+				},
+			},
+		},
+		{
+			ctx:   ctx,
+			input: []byte(`[{"balance":2214.189114310433,"change":-0.00255888,"currency":"usdt","text":"TCOM_USDT:263179103241933644","time":1755738516,"time_ms":1755738516430,"type":"fee","user":"12870774"}]`),
+			expected: []account.Change{
+				{
+					Account:   "12870774",
+					AssetType: asset.USDTMarginedFutures,
+					Balance: &account.Balance{
+						Currency:               currency.USDT,
+						Total:                  2214.189114310433,
+						Free:                   2214.189114310433,
+						AvailableWithoutBorrow: 2214.189114310433,
+						UpdatedAt:              time.UnixMilli(1755738516430),
+					},
+				},
+			},
+		},
+	} {
+		err := e.processBalancePushData(tc.ctx, tc.input, asset.USDTMarginedFutures)
+		if tc.err != nil {
+			require.ErrorIs(t, err, tc.err)
+			continue
+		}
+		require.NoError(t, err, "processBalancePushData must not error")
+		require.Len(t, e.Websocket.DataHandler, 1)
+		payload := <-e.Websocket.DataHandler
+		got, ok := payload.([]account.Change)
+		require.Truef(t, ok, "Expected account changes, got %T", payload)
+		checkAccountChange(t, tc.expected, got, e, tc.ctx)
+	}
+}
+
+func checkAccountChange(t *testing.T, expected []account.Change, received []account.Change, exch *Exchange, ctx context.Context) {
+	t.Helper()
+
+	require.Lenf(t, received, len(expected), "Expected %d changes, got %d", len(expected), len(received))
+	for i := range received {
```

Comment:
Feels like since we're in a test ...
```suggestion
	for i, got := range received {
		exp := expected[i]
```